brain/CMakeLists.txt

```diff
@@ -8,16 +8,22 @@ source_group(\\ FILES CMakeLists.txt)
 
 set(BRAIN_PUBLIC_HEADERS
   circuit.h
-  morphology.h
   types.h
+  cell/morphology.h
+  cell/section.h
+  cell/soma.h
```

Comment:
Very hard to say.
As it is now, I have a slight preference for the option 1, as having a "neuron" namespace makes the meaning of the classes very clear: brain::neuron::Morphology,  brain::neuron::Section, brain::neuron::Soma.
If Soma is not a nested class of Morphology, because the implementation would be ugly, then I have doubts about replacing the concept with an extra namespace.
With option 2, we will still have the name collision with a future glia::Morphology, unless it is combined with option 1 later as you also suggest. But then the hierarchy becomes quite long, and I am not sure if the "morphology" namespace brings much additional information in a brain::neuron::morphology::Soma.
(The last unknown is how a future glia morphology would be implemented. I am assuming here that they will be different enough and there will not be the use for a class hierarchy between the two...)